{{cookiecutter.project_name}}/notebooks/ModelValidation.py

```diff
@@ -0,0 +1,224 @@
+
+##################################################################################
+# TODO : Please fill in this cell with proper values for enable_baseline_comparison,
+#        data, targets, model_type, custom_metrics, validation_thresholds
+#        evaluator_config
+##################################################################################
+
+# Whether to load the current registered "Production" stage model as baseline. A version with "Production" stage must 
+# exist for the model.
+# Baseline model is a requirement for relative change and absolute change validation rules.
+# TODO(required) : enable_baseline_comparison
+enable_baseline_comparison = True
+
+# model validation data input for staging or prod workspace. A Pandas DataFrame or Spark DataFrame, containing evaluation features and labels.
+# Please refer to data parameter in mlflow.evaluate documentation https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate
+if env == "prod":
+    # TODO(required) : set data for prod workspace
+    data = None
+elif env == "staging":
+    # TODO(required) : set data for staging workspace
+    data = None
+else:
+    raise Exception("Unknown environment. Please select 'prod' or 'staging' as environment name")
+
+# The string name of a column from data that contains evaluation labels.
+# Call get_targets_from_recipe() to get targets from recipe configs if mlflow recipe is used. 
+# Please refer to targets parameter in mlflow.evaluate documentation https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate
+# TODO(required) : targets
+targets = get_targets_from_recipe()
+
+
+# A string describing the model type. The model type can be either "regressor" and "classifier".
+# Call get_model_type_from_recipe() to get model type from recipe configs if mlflow recipe is used. 
+# Please refer to model_type parameter in mlflow.evaluate documentation https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate
+# TODO(required) : model_type
+model_type = get_model_type_from_recipe()
+
+# Custom metrics to be included. Set it to None if custom metrics are not needed.
+# Please refer to custom_metrics parameter in mlflow.evaluate documentation https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate
+# TODO(optional) : define custom metric function as necessary
+# TODO(optional) : custom_metrics
+custom_metrics = []
+
+# Define model validation rules. Set it to None if validation rules are not needed.
+# Please refer to custom_metrics parameter in mlflow.evaluate documentation https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate
+# TODO(optional) : validation_thresholds
+validation_thresholds = {}
+
+# A dictionary of additional configurations to supply to the evaluator.
+# Please refer to evaluator_config parameter in mlflow.evaluate documentation https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate
+# TODO(optional) : evaluator_config
+evaluator_config = {}
```

Comment:
I'm wondering if these should be notebook arguments, what do you think?

Reply:
Do you mean the evaluator_config or all the parameters in general? 
For the evaluator_config we can just set it to None.
For the overall parameters, the custom_metrics and validation_thresholds can be huge blocks. It's better to have the stay in notebook or create separate notebook or yaml file.

Reply:
Agree with @mingyu89 that this doesn't need to be notebook argument. It's unlikely to change frequently.